### ISSUE TITLE
Why use symmetric padding ?

### ISSUE THREAD

**Issue by cipri-tom** [solivr/tf-crnn#9]
Hi,

I noticed that in the case of input images of different sizes, the padding is with `SYMMETRIC` values and only at the right side of the image ([code](https://github.com/solivr/tf-crnn/blob/9d12de194c8c1a935e2c63c84270dd11a62fb410/src/data_handler.py#L151)). Why not use the `CONSTANT` option, with `constant_values = 255` (or `1` if the image is normalised) ?
(edit: or `constant_values = avg_color` of the image)

Also, in the other case, when the image is less than half the `target_width`, it is replicated horizontally. However, the ground-truth data is not replicated. Does this not hurt the algorithm ?

Thank you !

**Comment by solivr** [comment]
Hi,

The `SYMMETRIC` padding is to avoid screwing up the statistics of the batch norm. I believe adding statically similar data when padding will add less 'side effects' than padding with `0`, `255` or `1`. However I hadn't confirmed it experimentally, so it may have no influence at all and you may be able to use `CONSTANT` padding.

Concerning the `target_width`, I want the the data to have the same size in order to simplify the batch-training. Since the 'real' width is kept in memory and passed to `tf.nn.ctc_loss` ([code](https://github.com/solivr/tf-crnn/blob/master/src/model.py#L268)) you do not pass the sequence of the additional replication (or better said, the `ctc_loss` doesn't take it into account). There might be small side effects on the borders when the convolution or max-pooling are performed (for example if the center of the kernel is in the border of the original image) but I don't think this is really a problem (but again I haven't formally checked).

**Comment by cipri-tom** [comment]
Ah, I didn't realise that the `ctc_loss` only takes the first `sequence_length` features into account. Then yes, your heuristics are very well grounded. 

Thank you for the explanation !
